client-skeleton-subscriber/src/main/java/eu/arrowhead/client/skeleton/subscriber/SubscriberMain.java

```diff
@@ -0,0 +1,120 @@
+package eu.arrowhead.client.skeleton.subscriber;
+
+import java.util.Base64;
+import java.util.Map;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.ApplicationArguments;
+import org.springframework.boot.ApplicationRunner;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.ComponentScan;
+
+import eu.arrowhead.client.library.ArrowheadService;
+import eu.arrowhead.client.library.util.ClientCommonConstants;
+import eu.arrowhead.client.skeleton.subscriber.constants.SubscriberConstants;
+import eu.arrowhead.common.CommonConstants;
+import eu.arrowhead.common.dto.shared.SystemRequestDTO;
+import eu.arrowhead.common.exception.InvalidParameterException;
+
+@SpringBootApplication
+@EnableConfigurationProperties(ConfigEventProperites.class)
+@ComponentScan(basePackages = {CommonConstants.BASE_PACKAGE}) //TODO: add custom packages if any
+public class SubscriberMain implements ApplicationRunner {
+
+    //=================================================================================================
+	// members
+	
+	@Value( SubscriberConstants.$PRESET_EVENT_TYPES_WD )
+	private String presetEvents;
+	
+	@Value(ClientCommonConstants.$CLIENT_SYSTEM_NAME)
+	private String clientSystemName;
+	
+	@Value(ClientCommonConstants.$CLIENT_SERVER_ADDRESS_WD)
+	private String clientSystemAddress;
+	
+	@Value(ClientCommonConstants.$CLIENT_SERVER_PORT_WD)
+	private int clientSystemPort;
+	
+	@Autowired
+	ArrowheadService arrowheadService;
+	
+	@Autowired
+	ConfigEventProperites configEventProperites;
+	
+	private final Logger logger = LogManager.getLogger(SubscriberApplicationInitListener.class);
+	
+	//=================================================================================================
+	// methods
+
+	//-------------------------------------------------------------------------------------------------
+	public static void main(final String[] args) {
+		SpringApplication.run(SubscriberMain.class, args);
+		
+	}
+
+	@Override
+	public void run(ApplicationArguments args) throws Exception {
+
+		subscribeToPresetEvents();
+		
+	}
+	
+	//=================================================================================================
+	// assistant methods
+
+	//-------------------------------------------------------------------------------------------------
+	private void subscribeToPresetEvents() {
+		
```

Comment:
Shoud have some comment here, that this is an example.... etc...

Reply:
fixed